src/v/cluster/jumbo_log_stm.h

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "bytes/iobuf.h"
+#include "jl/metadata.h"
+#include "model/fundamental.h"
+#include "model/record.h"
+#include "model/timeout_clock.h"
+#include "raft/persisted_stm.h"
+#include "raft/state_machine.h"
+#include "utils/mutex.h"
+#include "utils/uuid.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/util/log.hh>
+
+#include <absl/container/btree_map.h>
+#include <absl/strings/string_view.h>
+
+namespace config {
+struct configuration;
+}
+
+namespace cluster {
+
+class jumbo_log_stm final : public raft::persisted_stm<> {
+public:
+    static constexpr std::string_view name = "jumbo_log_stm";
+
+    explicit jumbo_log_stm(ss::logger&, raft::consensus*);
+
+    explicit jumbo_log_stm(
+      ss::logger&, raft::consensus*, config::configuration&);
+
+    ss::future<jumbo_log::write_intent_id_t> create_write_intent(
+      jumbo_log::segment_object, model::timeout_clock::duration);
+
+    // ss::future<stm_allocation_result>
+    // allocate_id(model::timeout_clock::duration timeout);
+
+    ss::future<iobuf> take_snapshot(model::offset) final { co_return iobuf{}; }
```

Comment:
Is it not implemented or it's not supposed to take snapshots?

Reply:
Just not implemented yet. It must take snapshots.